### PR TITLE
[D2IQ-71860] Ambassador Preview

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**
Feature

**What this PR does/ why we need it**:
Adds [Ambassador](https://getambassador.io) as an addon in KBA.

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-71860

**Does this PR introduce a user-facing change?**:
```release-note
* added Ambassador addon
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [x] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

**Review by GoelDeepak — CHANGES_REQUESTED** [review]
`Preview` isn't in the list of requirements. Ambassador will be a supported addon

**Comment by shaneutt** [comment]
> `Preview` isn't in the list of requirements. Ambassador will be a supported addon

This has been removed, ready for re-review. I pinged you in slack as I want to discuss this some before we merge.

**Comment by shaneutt** [comment]
We're going to need to include cert-manager updates which are inbound, so for the moment I'm going to close this but keep the branch, the intention being to re-open with cert-manager tests once we're unblocked.

**Comment by shaneutt** [comment]
We've decided to do a "Preview" version after all, sans cert-manager integration for the time being.

**Review by branden — APPROVED** [review]

**Review by GoelDeepak — APPROVED** [review]

File: test/ambassador_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{{
+							Name:  "backend",
+							Image: "docker.io/datawire/quote:0.4.1",
+							Ports: []corev1.ContainerPort{{
+								Name:          "http",
+								ContainerPort: 8080,
+							}}}}}}}}
+		deployment, err := cluster.Client().AppsV1().Deployments(ns).Create(context.TODO(), &testDeployment, metav1.CreateOptions{})
```
Comment (vespian) [inline test/ambassador_test.go:56]: Do we want to wait for deployment to finish before we continue?

Comment (shaneutt) [reply test/ambassador_test.go:56]: You're right that we probably do want to do this, I'll make an update for it.

**Comment by dkoshkin** [comment]
We probably need a better method, but for Calico we should atleast add this similar to `istio` https://github.com/mesosphere/konvoy/blob/master/clientapis/pkg/kubernetes/encoder/encoders.go#L20

**Pull request merged**